lib/foreman_tasks/cleaner.rb

```diff
@@ -138,8 +135,44 @@ def with_backup_file(backup_dir, file_name)
     end
 
     def delete_dynflow_plans(chunk)
-      dynflow_ids = chunk.find_all { |task| task.is_a? Task::DynflowTask }.map(&:external_id)
-      ForemanTasks.dynflow.world.persistence.delete_execution_plans({ 'uuid' => dynflow_ids }, batch_size, @backup_dir)
+      delete_dynflow_plans_by_uuid chunk.find_all { |task| task.is_a? Task::DynflowTask }.map(&:external_id)
+    end
+
+    def delete_dynflow_plans_by_uuid(uuids)
+      ForemanTasks.dynflow.world.persistence.delete_execution_plans({ 'uuid' => uuids }, batch_size, @backup_dir)
+    end
+
+    def delete_orphaned_dynflow_tasks
+      with_noop(orphaned_dynflow_tasks, 'orphaned execution plan') do |source, name|
+        with_batches(source, name) do |chunk|
+          delete_dynflow_plans_by_uuid chunk.select_map(:uuid)
+        end
+      end
+    end
+
+    # source must respond to :count and :limit
+    def with_noop(source, name, noop_message = nil)
+      if noop
+        say '[noop] ' + noop_message if noop_message
+        say "[noop] #{source.count} #{name}(s) would be deleted"
+      else
+        yield source, name
+      end
+    end
+
+    def with_batches(source, name)
+      start_tracking_progress(name, source.count)
```

Comment:
Currently, when having an empty set, I get:

```
About to remove 0 orphaned execution plan(s)
Deleted  orphaned execution plan(s)
```

I guess, in this case, we should just mention that no orphaned execution plans were found on go on.

Reply:
Hope this is better
```
2018-03-16T14:54:25 [I|for|] No tasks found, skipping.                                                                                                                                                                                      
2018-03-16T14:54:25 [I|for|] No orphaned execution plans found, skipping.                                                                                                                                                                   
```